### ISSUE TITLE
[Arch] Add option to Arch_Stairs for top tread level with landing/floor

### ISSUE THREAD

**Issue by Roy-043** [FreeCAD/FreeCAD#8513]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://github.com/FreeCAD/FreeCAD/issues/7050#issuecomment-1160193697

### Subproject(s) affected?

Arch

### Idea description

It would be useful if Arch_Stairs had an option for a top tread that is level with the landing or floor. In the Netherlands this is more or less standard for wooden and prefabricated concrete stairs (almost all concrete stairs are prefabricated in the Netherlands).

For prefabricated concrete stairs this tread has the same depth as the standard treads. For wooden stairs its depth is typically reduced (Nosing+RiserThichness+...).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by paullee0** [comment]
Just read this. 

@semhustej any comment?

**Comment by semhustej** [comment]
I agree that this would be useful. This is standard stairs solution.
With current stairs settings one has to do workarounds to achieve what the OP wants to achieve.

For reference Revit has a End with Riser settings, which differentiate this exact case:

![Image](https://github.com/user-attachments/assets/b6a5e8a6-a4df-447f-9c2e-ffc9287b732d)

**Comment by semhustej** [comment]
Note different results when `End with Riser` enabled/disabled combined with Riser and Tread thickness:

![Image](https://github.com/user-attachments/assets/7b501a25-1130-453d-9218-ac9f1a0ff1f4)


This is a section. Note that `End with Riser` settings also changes the overlap with the slab above because the top riser adds to the structure length.

`End with Riser` disabled:
![Image](https://github.com/user-attachments/assets/efb98608-c095-42b0-b710-3e9cbd1a4b82)

`End with Riser` enabled:

![Image](https://github.com/user-attachments/assets/b96b30cd-a8c7-45ea-9120-8954ff40da7c)


I think this would be also useful to implement. It is related to: https://github.com/FreeCAD/FreeCAD/issues/24456
There are lot of options and combinations of how to tackle the 'last step'.

**Comment by paullee0** [comment]
> also changes the overlap with the slab above because the top riser adds to the structure length.

Could you elaborate slightly on what you mean here? 

Thanks.

**Comment by semhustej** [comment]
> > also changes the overlap with the slab above because the top riser adds to the structure length.
> 
> Could you elaborate slightly on what you mean here?
> 
> Thanks.

I have circled the difference in the pictures bellow. Nothing is different between these two stairs except `End with Riser` setting. 

![Image](https://github.com/user-attachments/assets/bc3904f5-b8e6-4c15-abfe-6328c57afed7)

![Image](https://github.com/user-attachments/assets/9d04c017-338b-4f6a-ad48-22d881efbe17)
